cgi-bin/DW/BlobStore.pm

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/perl
+#
+# DW::BlobStore
+#
+# Meta storage API for storing arbitrary blobs of content by key.
+#
+# Authors:
+#     Mark Smith <mark@dreamwidth.org>
+#
+# Copyright (c) 2016 by Dreamwidth Studios, LLC.
+#
+# This program is free software; you may redistribute it and/or modify it under
+# the same terms as Perl itself.  For a copy of the license, please reference
+# 'perldoc perlartistic' or 'perldoc perlgpl'.
+#
+
+package DW::BlobStore;
+
+use strict;
+use v5.10;
+use Log::Log4perl;
+my $log = Log::Log4perl->get_logger( __PACKAGE__ );
+
+use DW::BlobStore::LocalDisk;
+use DW::BlobStore::MogileFS;
+use DW::BlobStore::S3;
+
+my $blobstores;
+
+sub _get_blobstores {
+    # If we've already created one, simply return it.
+    return $blobstores if defined $blobstores;
+
+    # Site owners must configure one of these. Disk is configured by default
+    # to point to our run directory which will work for test environs.
```

Comment:
Comment removed. I decided against that later. (Although we can ship a default config file with a setting like this, which might be worth doing.)